R/rdeephaven/src/client.cpp

```diff
@@ -0,0 +1,113 @@
+#include <memory>
+#include <exception>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "deephaven/client/client.h"
+#include "deephaven/client/flight.h"
+
+#include <arrow/c/abi.h>
+#include <arrow/c/bridge.h>
+
+#include <Rcpp.h>
+
+
+// ######################### DH WRAPPERS, API #########################
+
+class TableHandleWrapper {
+public:
+    TableHandleWrapper(deephaven::client::TableHandle ref_table) : internal_tbl_hdl(std::move(ref_table)) {};
+
+    TableHandleWrapper* select(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.select(columnSpecs));
+    };
+
+    TableHandleWrapper* view(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.view(columnSpecs));
+    };
+
+    TableHandleWrapper* dropColumns(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.dropColumns(columnSpecs));
+    };
+
+    TableHandleWrapper* update(std::vector<std::string> columnSpecs) {
+        return new TableHandleWrapper(internal_tbl_hdl.update(columnSpecs));
+    };
+
+    SEXP getStream() {
+        ArrowArrayStream* stream = new ArrowArrayStream;
```

Comment:
1) Your code is not minimizing the scope of variables.  This is not needed until the bottom of the function, but it is defined at the top.
2) `ArrowArrayStream* stream = new ArrowArrayStream();` would be more clear.

Reply:
Made the change, would like to know what difference `()` makes.